internal/gonut/cmd/push.go

```diff
@@ -166,61 +171,116 @@ func genericCommandFunc(cmd *cobra.Command, args []string) {
 	}
 }
 
-func runSampleAppPush(app sampleApp) error {
-	var cleanupSetting cf.AppCleanupSetting
+func mapDeleteSetting(deleteSetting string) (cf.AppCleanupSetting, error) {
+
 	switch deleteSetting {
 	case "always":
-		cleanupSetting = cf.Always
-
+		return cf.Always, nil
 	case "never":
-		cleanupSetting = cf.Never
+		return cf.Never, nil
 
 	case "on-success":
-		cleanupSetting = cf.OnSuccess
+		return cf.OnSuccess, nil
 
 	default:
-		return fmt.Errorf("unsupported delete setting: %s", deleteSetting)
+		return -1, fmt.Errorf("unsupported delete setting: %s", deleteSetting)
 	}
+}
 
-	appName := text.RandomStringWithPrefix(app.appNamePrefix, 32)
+func mapOutputSetting(outputSetting string) (cf.OutputType, error) {
 
-	directory, err := app.assetFunc()
-	if err != nil {
-		return err
+	switch strings.ToLower(outputSetting) {
+	case "json":
+		return cf.JSON, nil
+	case "yaml":
+		return cf.YAML, nil
+	default:
+		// return -1, fmt.Errorf("unsupported output type: %s", outputSetting)
+		return -1, nil
 	}
+}
 
-	report, err := cf.PushApp(app.caption, appName, directory, cleanupSetting)
-	if err != nil {
-		return err
-	}
+func summaryPrintout(appCaption string, report *cf.PushReport, summarySetting string, outputType cf.OutputType) error {
 
 	switch summarySetting {
-	case "quiet":
-		// Nothing to report
-
 	case "short", "oneline":
```

Comment:
For consistency, if the output is a separate flag (`--output`), then we have to make sure that also the short (oneliner) output supports the different output styles. It may look silly, but we would have to create an output JSON like this:
```json
{ "message": "Successfully pushed Staticfile sample app in 31 s." }
```